### ISSUE TITLE
List known validation errors/warnings

### ISSUE THREAD

**Issue by hannalaakso** [alphagov/govuk-frontend#1280]
## Description

Once we've finished the [WCAG 2.1 compliance work](https://github.com/alphagov/govuk-design-system/issues/677), we should consider increasing visibility of any known validation errors with our code due to accessibility, legacy etc fixes. This would help users to know which errors/warnings are known / expected.

An example would be the `role` attribute we add to `<header>`, `<footer>`, `<button>` for older browsers that don't support HTML5 elements, and the [`src` on SVG issue](https://github.com/alphagov/govuk-frontend/blob/master/src/components/header/template.njk#L33). 

## Checklist
- [x] Review and list existing / known issues -> [resulting list](https://github.com/alphagov/govuk-frontend/issues/1280#issuecomment-509588851) 
- [x] Triage [list](https://github.com/alphagov/govuk-frontend/issues/1280#issuecomment-509588851) with team to decide which ones need to be investigated / fixed and which ones should be documented in the Design System / somewhere as known issues.

**Comment by aliuk2012** [comment]
Linked to https://github.com/alphagov/govuk-design-system/issues/677

**Comment by hannalaakso** [comment]
WIP here: https://docs.google.com/spreadsheets/d/1bK127XvQJ72m8Xtn5006BKDmQlT5d-JIGM_a05w1hIA/

**Comment by hannalaakso** [comment]
This could be useful https://stevefaulkner.github.io/Articles/On%20HTML%20belts%20and%20ARIA%20braces%20(The%20Default%20Implicit%20ARIA%20semantics%20they%20didn%E2%80%99t%20want%20you%20to%20know%20about).html (thanks @leekowalkowski-hmrc)

**Comment by hannalaakso** [comment]
## Known validation errors / warnings

We need to investigate/document these.

| Component (s)                              | Message                                                                                              | Explanation / issue raised                                                                                                                                                                                                     | Flagged by                   |
|--------------------------------------------|------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------|
| Button - disabled                          | Attribute aria-disabled is unnecessary for elements that have attribute disabled                     |                                                                                                                                                                                                                                | https://validator.w3.org     |
| Checkboxes, radios with label as heading   | Element h1 not allowed as child of element legend in this context.                                   | Valid as of HTML 5.2 (https://www.w3.org/TR/html52/sec-forms.html#elementdef-legend), but living standard does not (yet?) recognise it as valid (https://html.spec.whatwg.org/multipage/form-elements.html#the-legend-element) | https://validator.w3.org     |
| Checkboxes, radios with conditional reveal | Elements must only use allowed ARIA attributes                                                       | https://github.com/alphagov/govuk-frontend/issues/979                                                                                                                                                                          | Axe Chrome extension         |
| Checkboxes, radios with conditional reveal | Radio inputs with the same name attribute value must be part of a group                              | https://github.com/alphagov/govuk-frontend/issues/979                                                                                                                                                                          | Axe Chrome extension         |
| Date input, details                        | A group created with role=group is missing a name                                                    | https://github.com/alphagov/govuk-frontend/blob/master/src/govuk/components/date-input/template.njk#L80                                                                                                                        | Siteimprove Chrome extension |
| Date input                                 | Attribute pattern is only allowed when the input type is email, password, search, tel, text, or url. | https://github.com/alphagov/govuk-frontend/issues/1449                                                                                                                                                                         | https://validator.w3.org     |
| Date input                                 | The group role is unnecessary for element fieldset.                                                  |                                                                                                                                                                                                                                | https://validator.w3.org     |
| Footer                                     | The contentinfo role is unnecessary for element footer                                               |    For browsers that don't support [HTML5](https://caniuse.com/#feat=html5semantic). Confirm if still needed.                                                                                                                                                                                                                           | https://validator.w3.org     |
| Header                                     | The banner role is unnecessary for element header.                                                   |    For browsers that don't support [HTML5](https://caniuse.com/#feat=html5semantic). Confirm if still needed.                                                                                                                                                                                                                             | https://validator.w3.org     |
| Header                                     | Attribute src not allowed on element image at this point.                                            | https://github.com/alphagov/govuk-frontend/blob/b57ade99ac353a944620d818c1f89e1005935181/src/govuk/components/header/template.njk#L34                                                                                          | https://validator.w3.org     |
| Header                                     | Possible misuse of aria-label                                                                        | https://github.com/alphagov/govuk-frontend/issues/1340                                                                                                                                                                         | https://validator.w3.org     |
| Header                                     | The button role is unnecessary for element button.                                                   |                                                                                                                                                                                                                                | https://validator.w3.org     |
| Button - disabled link                     | Color contrast is insufficient                                                                       | https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html                                                                                                                                                             | Axe Chrome extension         |
| Button - disabled warning                  | Color contrast is insufficient                                                                       | https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html                                                                                                                                                             | Siteimprove Chrome extension |
|                                            |                                                                                                      |                                                                                                                                                                                                                                |                              |

**Comment by 36degrees** [comment]
Added a new entry to the table for details:

`<dl> elements must only directly contain properly-ordered <dt> and <dd> groups, <script> or <template> elements`

HTML 5.2 permits `<div>` as a descendant of `<dl>` – previous versions do not.

Thanks to @MalcolmVonMoJ for flagging.

**Comment by nickcolley** [comment]
When we complete this we can document and close the conditional reveal related issues:

- https://github.com/alphagov/govuk-frontend/issues/979
- https://github.com/alphagov/govuk-frontend/issues/989
- https://github.com/alphagov/govuk-frontend/issues/935

**Comment by hannalaakso** [comment]
Thanks @nickcolley. I've documented https://github.com/alphagov/govuk-frontend/issues/979 as it's a warning. https://github.com/alphagov/govuk-frontend/issues/979 and https://github.com/alphagov/govuk-frontend/issues/935 are bugs, I don't think we meant to document those within this particular card.

**Comment by hannalaakso** [comment]
Closing as we're going to publish the list as documentation as part of https://github.com/alphagov/govuk-design-system/issues/1048

**Comment by hannalaakso** [comment]
Reopening this as users might search for these errors/warnings under open issues in this repo. Closed when https://github.com/alphagov/govuk-frontend/issues/1687 is done.

**Comment by nickcolley** [comment]
Would be good to consider getting this published properly at some point, had reports from a user that it's not very easy to use as a reference right now.

> With Hanna’s list being inside a comment, it isn’t especially easy to read (scrolling horizontally is a pain) and it’d be good to surface the reported bugs, as I didn’t know where to look for them (apologies if there is a big link somewhere obvious :slightly_smiling_face: ).

**Comment by hannalaakso** [comment]
As discussed above the table is hard to use, especially with regards to horizontal scrolling. I've tried to add some inline styles / HTML to the table to limit the column widths to reduce the need to scroll horizontally but Github markdown strips out any inline styles / `col` attributes so I don't see a way to control the column widths. We're basically asking too much of the Github table format and should look to publish these differently.

**Comment by jpveooys** [comment]
Hi,
The above list seems to be missing https://github.com/alphagov/govuk-frontend/issues/2472.

**Comment by 36degrees** [comment]
All of these issues are now listed in a [public GitHub Project](https://github.com/orgs/alphagov/projects/37/views/1) – I plan to raise a pull request to update links to this issue to point to the new project and then close this issue.
